docs/infra/attestation-provider.md

```diff
@@ -0,0 +1,155 @@
+There are two situations in which it makes sense to do so:
+
+* Running unincentivized local attestation providers to act as a safeguard against malicious forks,
+* Running incentivized attestation providers to reap rewards, contributing to and helping improve the decentralization of the network while providing attestations for the [state connector](#state-connector).
+
+!!! note "Rewards"
+
+    Currently, the process of rewarding public attestation providers is still being discussed.
+
+## Prerequisites
+
+Before proceeding with the [installation](#installation), it is essential to follow every prerequisite.
+
+### System Requirements
+
+The following specifications were observed to be able to run nodes for all supported blockchains on a single machine.
+Ideally, a dedicated node should be run for each connected blockchain in production for better performance and uptime.
+
+* **OS**: Ubuntu 20.04
+* **CPU Cores**: 16
+* **RAM**: 96GB
+* **Disk space**: 3TB with an option or plan in place to expand capacity. SSD is recommended because some chains like xrpl can fall out of sync on regular disks.
+
+As of Q2 2022, this is roughly what you can expect from each node regarding disk usage:
+
+| Volume              | Size  |
+|---------------------|-------|
+| algorand-data       | 1TB   |
+| algorand-indexer-db | 1TB   |
+| bitcoin-data        | 500GB |
+| dogecoin-data       | 60GB  |
+| litecoin-data       | 90GB  |
+| ripple-data         | 230GB |
+
+### Dependencies
+
+* [`git`](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [`docker`](https://docs.docker.com/get-docker/)
+* [`docker-compose`](https://docs.docker.com/compose/install/)
+* [`openssl`](https://www.openssl.org/source/)
+* [`python3`](https://www.python.org/downloads/)
+
+### Node Setup
+
+The first step is to set up a node for each supported network, which the attestation provider suite can then use to make attestations.
+
+The supported chains at the moment are the following:
+
+* [Bitcoin](https://github.com/bitcoin/bitcoin)
+* [Litecoin](https://github.com/litecoin-project/litecoin)
+* [Dogecoin](https://github.com/dogecoin/dogecoin)
+* [Rippled](https://github.com/ripple/rippled)
+* [Algorand](https://github.com/algorand/go-algorand)
+
+!!! warning "Setting the nodes up is a very long process"
+
+    Since the attestation provider suite requires having access to all supported networks, it is required to have an up-to-date node setup for each network.
+    Certain networks take very long times to get synchronized with, such as Algorand which requires indexing the whole history of the blockchain locally.
+    In our testing it took a few hours to sync up with Litecoin, Dogecoin and XRPL, more than a day for Bitcoin and weeks for Algorand.
+
+#### Chain Configuration
+
+Before the nodes can be launched, they have to be configured with credentials to access their respective networks.
```

Comment:
I'm OK with using the script, even if it's badly named, as long as we clearly explain what it does. So make sure you explicitly say that the script just generates credentials and writes them to config files. Maybe even say that the following steps will take care of downloading/running/whatever.

Also, you can try to ask the devs to use a better suited name. They'll say this was just for internal use, never intended to go public, etc, etc, but then you answer that it is going public now because you're writing a guide and it would benefit everybody to use a better name like `config_credentials.sh` or whatever 😄